packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdk.kt

```diff
@@ -285,6 +304,44 @@ class DdSdk(
         }
     }
 
+    private fun buildVitalUpdateFrequency(vitalsUpdateFrequency: String?): VitalsUpdateFrequency {
+        val vitalUpdateFrequencyLower = vitalsUpdateFrequency?.lowercase(Locale.US)
+        return when (vitalUpdateFrequencyLower) {
+            "never" -> VitalsUpdateFrequency.NEVER
+            "rare" -> VitalsUpdateFrequency.RARE
+            "average" -> VitalsUpdateFrequency.AVERAGE
+            "frequent" -> VitalsUpdateFrequency.FREQUENT
+            else -> VitalsUpdateFrequency.AVERAGE
+        }
+    }
+
+    private fun monitorJsRefreshRate(vitalsUpdateFrequency: VitalsUpdateFrequency) {
+        val frameTimeCallback = buildFrameTimeCallback(vitalsUpdateFrequency)
+        reactContext.runOnJSQueueThread {
+            val vitalFrameCallback =
+                VitalFrameCallback(frameTimeCallback) { isInitialized() }
+            try {
+                Choreographer.getInstance().postFrameCallback(vitalFrameCallback)
+            } catch (e: IllegalStateException) {
+                // This should never happen as the React Native thread always has a Looper
+                val errorMessage = e.message
+                datadog.telemetryError(errorMessage ?: MONITOR_JS_ERROR_KIND, e.stackTraceToString(), MONITOR_JS_ERROR_KIND)
```

Comment:
There's a version of `error` in `Telemetry` that takes a `Throwable`, does it make more sense to use that instead in this situation?

Reply:
Yes I think so, thanks for the advice!